api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -266,6 +294,22 @@ public <T> R notStartsWith(Bound<T> expr, Literal<T> lit) {
       throw new UnsupportedOperationException("Unsupported operation.");
     }
 
+    public <T> R endsWith(Bound<T> expr, Literal<T> lit) {
+      throw new UnsupportedOperationException("Unsupported operation.");
```

Comment:
Could you include the relevant details in these exception messages? "endsWith is an unsupported operation" etc.